### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by almirsarajcic** [3Stones-io/omedis#310] [General]
To translate new strings, we use Cursor in combination with ChatGPT. We select one or multiple .po files of the same language to provide context to the ChatGPT and then ask it to add missing translations. 

<img width="465" alt="Screenshot 2024-11-09 at 17 53 00" src="https://github.com/user-attachments/assets/c5280841-22e6-4654-94af-0c867ca3a10c">

But to get to that point, we first need to extract all strings to be translated.
We do that with the `mix gettext.extract --merge` command.
If we get into an issue running that command (bad merge, merge conflicts...), we can get the latest working version (with some git magic ✨) and rerun the command.

If something really bad happened we can go for a desperate measure and remove the `priv/gettext` directory, then regenerate again:
```bash
rm -rf priv/gettext
mkdir -p priv/gettext/{de,en,fr,it}/LC_MESSAGES
mix gettext.extract
mix gettext.merge priv/gettext
```

**Comment by fradarr** [comment]
Great stuff, this one is for the Wiki page as well, we'll discuss it today!
